core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -474,6 +479,8 @@ private[kafka] class Processor(val id: Int,
             trace("Closing socket connection actively according to the response code.")
             close(selector, curr.request.connectionId)
         }
+      } catch {
+        case NonFatal(e) => processException(s"Exception while processing response for ${curr.request.connectionId}", e)
```

Comment:
Should we close the connection here?

Reply:
@junrao Thank you for the review. Added `close`.